Problem5/test/problem5/Problem5Test.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package problem5;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+/**
+ *
+ * @author User
+ */
+public class Problem5Test {
+    
+    private static String[] words = null; //List of all words
+    private static int[] indexes = null; //Indexes of the first wrods to start with each letter
+    
+    @BeforeClass
+    public static void setUpClass()
+    {
+        words = new String[4]; //Create array of words
+        words[0] = "bach";
+        words[1] = "evan";
+        words[2] = "joao";
+        words[3] = "louie";
+        indexes = new int[26]; //Set indexes of starting letters
+        int indexNum = 0;
+        while (indexNum < 2)
+        {
+            indexes[indexNum] = 0;
+            indexNum++;
+        }
+        while (indexNum < 5)
+        {
+            indexes[indexNum] = 1;
+            indexNum++;
+        }
+        while (indexNum < 10)
+        {
+            indexes[indexNum] = 2;
+            indexNum++;
+        }
+        while (indexNum < 26)
+        {
+            indexes[indexNum] = 3;
+            indexNum++;
+        }
+    }
+    
+    /**
+     * Test of binarySearch method, of class Problem5.
+     */
+    @Test
+    public void testBinarySearch() {
```

Comment:
Your testing should include edge cases like empty arrays and arrays of a single element.